metadata/src/utils/validation.rs

```diff
@@ -588,7 +584,7 @@ impl<'a> MetadataHasher<'a> {
 
         // Get the hashes of outer enums, considering only `specific_pallets` (if any are set).
         // If any of the typed that represent outer enums are encountered later, hashes from `top_level_enum_hashes` can be substituted.
-        let outer_enum_hashes = OuterEnumHashes::new(metadata, self.specific_pallets.as_deref());
+        let outer_enum_hashes = OuterEnumHashes::new(metadata);
```

Comment:
So here we no logner care about the possibility of the enums like `RuntimeCall` etc being changed, but how can we avoid caring about this?

(either the variants and all of their types still exist, so the hashes are the same, or we have stripped some types and/or variants and then the hashes owuld be different unless we ignore them, wouldn't they?)

Reply:
list of pallets/variants types is completely dependant on the given set of pallets for a particular pallet or unpredictable

Reply:
I don't understand your comment Pavlo, do you mean that one can't really filter out enum variant based on pallet names because it may be referenced by that some pallet may depend on another pallet or some type from that pallet?

Reply:
Aight, I think this was opinionated and removed variants without checking that it was actually referenced by pallets, runtime apis?

We now only rely on TypeSet::retain_metadata(..) to achieve this